crates/circuit/src/packed_instruction.rs

```diff
@@ -628,3 +715,23 @@ impl PackedInstruction {
         }
     }
 }
+
+// When cloning a `PackedInstruction` we will need to delete the old cached
+// `Gate` to avoid having duplicated references to the same python object.
+impl Clone for PackedInstruction {
+    fn clone(&self) -> Self {
+        Self {
+            op: self.op.clone(),
+            qubits: self.qubits,
+            clbits: self.clbits,
```

Comment:
These don't have to be cloned because they are interned, right?

Reply:
They don't have to be cloned because I believe they implement the `Copy` trait, so it's just a matter of re-using the bits, no need to re-build them.